src/main/java/com/exadel/discountwebapp/location/controller/LocationController.java

```diff
@@ -0,0 +1,53 @@
+package com.exadel.discountwebapp.location.controller;
+
+import com.exadel.discountwebapp.location.service.LocationService;
+import com.exadel.discountwebapp.location.vo.LocationRequestVO;
+import com.exadel.discountwebapp.location.vo.LocationResponseVO;
+import lombok.AllArgsConstructor;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/api/locations")
+@AllArgsConstructor
+public class LocationController {
+    private LocationService locationService;
+
+    @GetMapping
+    public List<LocationResponseVO> getAllLocations() {
+        return locationService.findAll();
+    }
+
+    @GetMapping("/{id}")
+    public LocationResponseVO getLocationById(@PathVariable Long id) {
+        return locationService.findById(id);
+    }
+
+    @GetMapping("/country")
+    @ResponseBody
```

Comment:
no need to use @ResponseBody here
a good question is why?

Reply:
because parameters are sent via URL, not body